src/utils/gui/globjects/GUIPolygon.cpp

```diff
@@ -119,7 +118,12 @@ TesselatedPolygon::drawTesselation(const PositionVector& shape) const {
     if (myTesselation.empty()) {
         myCurrentTesselated = this;
         // draw the tesselated shape
-        double* points = new double[shape.size() * 3];
+        size_t shapeSize = shape.size() * 3;
+        size_t holesSize = 0;
+        for (PositionVector hole : myHoles) {
+            holesSize += hole.size() * 3;
+        }
+        double* points = new double[shapeSize + holesSize];
```

Comment:
I would prefer if you use just one numPoints counter here and if you iterate over the holes with a const reference

Reply:
> I would prefer if you use just one numPoints counter here and if you iterate over the holes with a const reference

I corrected that.